yjit/src/backend/ir.rs

```diff
@@ -376,6 +376,13 @@ pub struct Insn
     pub(super) pos_marker: Option<PosMarkerFn>,
 }
 
+impl Insn {
+    pub(super) fn get_out(&self) -> Opnd {
+        assert_ne!(self.out, Opnd::None, "No output operand allocated for this instruction. Please ensure you are using its output.");
+        self.out
+    }
+}
```

Comment:
I support the idea, but can we just change the `.into()` methods to output this message for `Opnd::None` instead, it would add less syntax noise and make it so you always get this message, can't forget to call `get_out()`.